app/graphql/resolvers/top_volunteer_resolver.rb

```diff
@@ -0,0 +1,79 @@
+module TopVolunteerResolver
+  class << self
+    def all(_object, args, context)
+      user_entry = '"users"."id", "users"."first_name", "users"."last_name", "users"."email", "users"."group", "users"."photo"'
+
+      events_scope = User.joins(:events).select("#{user_entry}, \"events\".\"duration\"")
```

Comment:
is there a reason why we are using string interpolation here?

Reply:
`user_entry` is used in multiple locations to denote the fields we actually want from `User`. Instead of repeating the same user fields in different parts of the code, putting it in a variable and using string interpolation reduces the duplication of the string literals that we'd otherwise have.

Reply:
Did you actually mean to ask why we're not using something like: `User.joins(:events).select(:id, :first_name)`?

That's a good point. I wasn't too aware of the usage of symbols here. It looks like it does work, but the last `:duration` would not translate to `"events"."duration"` in SQL.

Right now it will translate to just `"duration"`, which can be ambiguous. How would I reference it so it translates nicely to `"events"."duration"`?

Reply:
I answered my own question. Just need to use `:'events.duration'`.

Reply:
Bumped into a problem further down and I don't know how to solve it.
`all_events_scope = all_events_scope.select("#{user_entry},  SUM(\"duration\") AS \"duration\"").group(user_entry)`

I'm not sure how we can convert the `SUM()` here into something that is compatible with symbols.

Reply:
yeah ok, I am not familiar with active record, so I can't provide in-depth help here sorry. I was just worried about opening holes for sql injection. However after our chat, you have convinced me that the parts that are not using symbols are in our control.